internal/services/runservice/common/common.go

```diff
@@ -127,6 +128,8 @@ func GenExecutorTaskSpecData(r *types.Run, rt *types.RunTask, rc *types.RunConfi
 	// run config Environment variables ovverride every other environment variable
 	mergeEnv(environment, rc.Environment)
 
+	environment["AGOLA_RUN_COUNTER"] = strconv.FormatUint(r.Counter, 10)
```

Comment:
This is a special case since it's the unique env var generated at runtime and not saved in the runconfig. It should be documented the rationale behind this choice.